Parent/Parent/Create Account/CreateAccountViewController.swift

```diff
@@ -218,6 +249,30 @@ class CreateAccountViewController: UIViewController, ErrorViewController {
         termsAndConditionsTextView.sizeToFit()
         termsAndConditionsTextViewHeight.constant = termsAndConditionsTextView.frame.size.height
     }
+
+    func addPairingCodeToCurrentAccount(_ code: String) {
+        assert(presentingViewController is ErrorViewController, "presentingViewController needs to conform to ErrorViewController protocol")
+        guard let presentingVC = presentingViewController as? ErrorViewController else { return }
+        addStudentPairingController = AddStudentController(presentingViewController: self) { [unowned self] error in
+            AppEnvironment.shared.router.dismiss(self) {
+                if let error = error { presentingVC.showError(error) }
+            }
```

Comment:
Rather than assert some other controller can show the error, the typical course of action here is to not dismiss self & have it show the error. Only dismiss self if it completed successfully.